### PR TITLE
Riak CS 2.0.0 Release Updates 

### PR DESCRIPTION
**Pull request by TheOnlyRew**

### CODE REVIEW

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md [hunk 1]
```diff
@@ -156,126 +168,132 @@ The JSON response will look something like this:
...
-{admin_secret, "5BE84D7EEA1AEEAACF070A1982DDA74DA0AA5DA7"},
+Once the admin user exists, you must specify the credentials of the admin user
+on each node in the Riak CS system. The admin user credential settings reside in
+the Riak CS `riak-cs.conf` file, which is located in the `/etc/riak-cs`
+directory. Paste the `key_id` string between the quotes for the `admin.key`.
+Paste the `key_secret` string into the `admin.secret` variable, as shown here:
+
+```riakcsconf
+admin.key, 324ABC0713CD0B420EFC086821BFAE7ED81442C},
+admin.secret, 5BE84D7EEA1AEEAACF070A1982DDA74DA0AA5DA7},
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: Should this be ...

```
admin.key = 324ABC0713CD0B420EFC086821BFAE7ED81442C
admin.secret = 5BE84D7EEA1AEEAACF070A1982DDA74DA0AA5DA7
```

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md [hunk 2]
```diff
@@ -156,126 +168,132 @@ The JSON response will look something like this:
...
-creation limit, they are still able to perform other actions, including
-bucket deletion. You can change the default limit using the
-`max_buckets_per_user` parameter in each node's `app.config` file. The
-example configuration below would set the maximum to 1000:
+If you wish, you can limit the number of buckets created per user. The default
+maximum is 100. Please note that if a user exceeds the bucket creation limit,
+they are still able to perform other actions, including bucket deletion. You can
+change the default limit using the `max_buckets_per_user` parameter in each
+node's `advanced.config` file---there is no equivalent configuration for the
+`riak-cs.conf` file. The example configuration below would set the maximum to
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: Oh, this might be mistake: we should have this in riak-cs.conf...

Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: https://github.com/basho/riak_cs/issues/1109

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md [hunk 3]
```diff
@@ -383,26 +407,12 @@ Riak CS are available in [[Garbage Collection]].
...
-There are two configuration options designed to provide improved
-performance for Riak CS when using Riak 1.4.0 or later. These options
-take advantage of additions to Riak that are not present prior to
-version 1.4.0.
+
+<div class="note"><div class="title">Deprecated Configurations</div>
+While Riak CS 2.0.0 still allows the configuration of `gc_paginated_indexes` and
+`gc_batch_size`, it is strongly recommended that these settings not be used. The
+have been deprecated, and <em>will be removed</em> in the next major release.
+</div>
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: Ooops, I didn't intend deprecating `gc_batch_size` - just `gc_paginated_indexes` . Opened https://github.com/basho/riak_cs/issues/1109 .

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md [hunk 4]
```diff
@@ -54,10 +66,29 @@ following line:
...
+```advancedconfig
+{riak_kv, [
+    %% Other configs
+    {add_paths, ["/usr/lib/riak-cs/lib/riak_cs-{{VERSION}}/ebin"]},
+    {storage_backend, riak_cs_kv_multi_backend},
+    {multi_backend_prefix_list, [{<<"0b:">>, be_blocks}]},
+    {multi_backend_default, be_default},
+    {multi_backend, [
+        {be_default, riak_kv_eleveldb_backend, [
+            {max_open_files, 50},
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md]: Regarding what we described in release notes, I'd like this to be changed to `total_leveldb_mem_percent`. The value is ... say, 30?

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md [hunk 5]
```diff
@@ -157,32 +206,45 @@ If the response is `pong`, then Riak is running; if the response is
...
 ]}
 ```
 
-**Note**: The `pb` values in the Riak `app.config` file must match the
-values for `riak_ip` and `riak_pb_port` in the Riak CS and Stanchion
-`app.config` files.
+**Note**: The `listener.protobuf.internal` values in the Riak `riak.conf` (or
+the `pb` value in `advanced.config`/`app.config`) file must match the values for
+`riak_host`in the Riak CS and Stanchion `riak-cs.config` (or `riak_ip` and
+`riak_pb_port` in `advanced.config`/`app.config`) files.
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md]: I should have apologised that here old behaviour is not preserved. In `advanced.config`, the item is `{riak_host, {localhost, 8087}},` .

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Stanchion.md [hunk 6]
```diff
@@ -8,60 +8,94 @@ audience: intermediate
...
 
 ```appconfig
 {stanchion, [
-
-    %% Configs here
-
-]}
+             {stanchion_ip, "127.0.0.1"},
+             {stanchion_port, 8080}
+             %% Other configs
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Stanchion.md]: ditto. `{host, {"127.0.0.1", 8080}}` here.

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md [hunk 7]
```diff
@@ -70,68 +78,72 @@ the local host. If your Riak CS system has multiple nodes, however, you
...
+stanchion_host = 127.0.0.1:8085
+stanchion_ssl = on
+```
 
-```appconfig
+```advancedconfig
 {riak_cs, [
     %% Other configs
-
     {stanchion_ip, "127.0.0.1"},
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: `stanchion_ip` and `stanchion_port` is not preserved. `{host, {"127.0.0.1", 8085}}` should be used.

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Stanchion.md [hunk 8]
```diff
@@ -8,60 +8,94 @@ audience: intermediate
...
-* `riak_pb_port` --- Replace `8087` with the port number set in the Riak
-  `app.config` file
+```stanchionconf
+riak_host = 127.0.0.1:8087
+```
+
+```appconfig
+{stanchion, [
+             {riak_ip, "127.0.0.1"},
+             {riak_pb_port, 8087}
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Stanchion.md]: ditto ... These are changed to `{riak_host, {"127.0.0.1", 8087}}" even in `app.config` .

File: source/languages/en/riakcs/tutorials/fast-track/Building-a-Local-Test-Environment.md [hunk 9]
```diff
@@ -162,112 +162,101 @@ package you are installing.
...
+buckets.default.allow_mult = true
+```
+
+Next, you need to expose the necessary Riak CS modules to Riak and instruct Riak
+to use the custom backend provided by Riak CS. You'll have to use the old-style
+`/etc/riak/advanced.config` for these settings. The file should look like:
+
+```advancedconfig
+[
+ {riak_core, [
```
Comment (kuenishi) [inline source/languages/en/riakcs/tutorials/fast-track/Building-a-Local-Test-Environment.md]: s/riak_core/riak_kv/

File: source/languages/en/riakcs/tutorials/fast-track/Building-a-Local-Test-Environment.md [hunk 10]
```diff
@@ -162,112 +162,101 @@ package you are installing.
...
+              {add_paths, ["/usr/lib/riak-cs/lib/riak_cs-{{VERSION}}/ebin"]},
+              {storage_backend, riak_cs_kv_multi_backend},
+              {multi_backend_prefix_list, [{<<"0b:">>, be_blocks}]},
+              {multi_backend_default, be_default},
+              {multi_backend, [
+                  {be_default, riak_kv_eleveldb_backend, [
+                      {max_open_files, 50},
```
Comment (kuenishi) [inline source/languages/en/riakcs/tutorials/fast-track/Building-a-Local-Test-Environment.md]: Let's change this `total_leveldb_mem_percent` as well as above part I pointed.

**Comment by kuenishi** [comment]
Amazing work just in two days! Thank you. And also, thank you for fixing my broken English in CS 2.0 release notes @TheOnlyRew .

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md (hunk 2 above)
Comment (TheOnlyRew) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: I just want to confirm, these setting are missing from 2.0.0, but will be added to 2.0.1?

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md (hunk 3 above)
Comment (TheOnlyRew) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: Thanks for the clarification. I'll change the wording in this section.

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md (hunk 1 above)
Comment (TheOnlyRew) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: Great catch, thank you.

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md (hunk 4 above)
Comment (TheOnlyRew) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md]: That's my mistake, I missed that. I'll change the name and the value.

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md (hunk 5 above)
Comment (TheOnlyRew) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md]: I'll make the appropriate changes. Thanks for point this out!

**Comment by TheOnlyRew** [comment]
Thanks for this last review to catch my remaining mistakes.
Great job on the code and the release notes---I just did my best to make it look pretty.

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md (hunk 4 above)
Comment (TheOnlyRew) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md]: Oh, another clarification, please. These settings that have changed in `advanced.config` -- `max_open_files`, `riak_host`, etc. -- should also be changed in `app.config`, correct?

**Comment by TheOnlyRew** [comment]
Thanks again for the review, @kuenishi. I think I changed everything correctly, but could you one more look at the following files/line numbers just to make sure that I've gotten it right?
Thanks!
Configuring-Riak.md:102
Configuring-Riak-CS.md:105
Configuring-Stanchion.md:52
Configuring-Stanchion.md:96

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md (hunk 2 above)
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: Yes, will do :)

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md (hunk 4 above)
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md]: Yes for both. But not changing `max_open_files` does not harm afaik.

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md [hunk 11]
```diff
@@ -70,68 +78,71 @@ the local host. If your Riak CS system has multiple nodes, however, you
...
+```
 
-```appconfig
+```advancedconfig
 {riak_cs, [
     %% Other configs
-
-    {stanchion_ip, "127.0.0.1"},
-    {stanchion_host, 8085},
+    {stanchion_host, {"127.0.0.1", 8085}}
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak-CS.md]: Oh, you need comma here as there's a next line. Should out to erlang term format.

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md [hunk 12]
```diff
@@ -68,7 +99,7 @@ to insert the following section:
     {multi_backend_default, be_default},
     {multi_backend, [
         {be_default, riak_kv_eleveldb_backend, [
-            {max_open_files, 50},
+            {total_leveldb_mem_percent, 30},
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Riak.md]: Correct.

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Stanchion.md [hunk 13]
```diff
@@ -8,60 +8,92 @@ audience: intermediate
...
+listener = 127.0.0.1:8080
+```
 
 ```appconfig
 {stanchion, [
-
-    %% Configs here
-
-]}
+             {host, {"127.0.0.1", 8080}},
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Stanchion.md]: Correct. Maybe 8080 is also used for CS default listening port, so 8085 might be more suitable?

File: source/languages/en/riakcs/cookbooks/configuration/Configuring-Stanchion.md [hunk 14]
```diff
@@ -8,60 +8,92 @@ audience: intermediate
...
-* `riak_ip` --- Replace `127.0.0.1` with the IP address of the Riak node
-* `riak_pb_port` --- Replace `8087` with the port number set in the Riak
-  `app.config` file
+```stanchionconf
+riak_host = 127.0.0.1:8087
+```
+
+```appconfig
+{stanchion, [
+             {riak_host, {"127.0.0.1", 8087}},
```
Comment (kuenishi) [inline source/languages/en/riakcs/cookbooks/configuration/Configuring-Stanchion.md]: Correct.

**Comment by kuenishi** [comment]
I checked all updates and they look nice except following comma lacking in Configuring-Riak-CS.md:105.

**Comment by TheOnlyRew** [comment]
Thanks again for the review, @kuenishi!

**Pull request merged**